src/core/obj.js

```diff
@@ -252,6 +254,114 @@ var Catalog = (function CatalogClosure() {
       }
       return dest;
     },
+
+    get pageLabels() {
+      var obj = null;
+      try {
+        obj = this.readPageLabels();
+      } catch (ex) {
+        if (ex instanceof MissingDataException) {
+          throw ex;
+        }
+        warn('Unable to read page labels.');
+      }
+      return shadow(this, 'pageLabels', obj);
+    },
+    readPageLabels: function Catalog_readPageLabels() {
+      var xref = this.xref;
+      var pageLabels = [];
+      var obj = this.catDict.getRaw('PageLabels');
+      if (obj) {
+        var style = null;
+        var prefix = '';
+        var start = 1;
+
+        var numberTree = new NumberTree(obj, xref);
+        var nums = numberTree.getAll();
+        var currentLabel = '', currentIndex = 1;
+
+        for (var i = 0, ii = this.numPages; i < ii; i++) {
+          if (nums.hasOwnProperty(i)) {
+            var labelDict = nums[i];
+            assert(isDict(labelDict), 'The PageLabel is not a dictionary.');
+
+            var type = labelDict.get('Type');
+            assert(!type || (isName(type) && type.name === 'PageLabel'),
+                   'Invalid type in PageLabel dictionary.');
+
+            var s = labelDict.get('S') || Name.get('');
+            assert(isName(s), 'Invalid style in PageLabel dictionary.');
+            style = s.name;
+
+            prefix = labelDict.get('P') || '';
+            assert(isString(prefix), 'Invalid prefix in PageLabel dictionary.');
+
+            start = labelDict.get('St') || 1;
+            assert(isInt(start), 'Invalid start in PageLabel dictionary.');
+            currentIndex = start;
+          }
+
+          switch (style) {
+            case 'D':
+              currentLabel = '' + currentIndex;
+              break;
+            case 'R':
+            case 'r':
+              var ROMAN_NUMBERS = [
```

Comment:
I think we should place this in the `Util` module (much like the annotation type constants) to avoid recreating this array each time a label is processed. Placing it in a utility module also seems like a better place for this kind of static data to me.